src/components/RelicModal.tsx

```diff
@@ -208,6 +211,26 @@ export default function RelicModal(props: {
   }
   const handleOk = () => {
     relicForm.submit()
+    const currentValue = {
+      grade: relicForm.getFieldValue('grade'),
+      enhance: relicForm.getFieldValue('enhance'),
+      part: relicForm.getFieldValue('part'),
+      set: relicForm.getFieldValue('set'),
+      mainStatType: relicForm.getFieldValue('mainStatType'),
+      substatType0: relicForm.getFieldValue('substatType0'),
+      substatValue0: relicForm.getFieldValue('substatValue0'),
+      substatType1: relicForm.getFieldValue('substatType1'),
+      substatValue1: relicForm.getFieldValue('substatValue1'),
+      substatType2: relicForm.getFieldValue('substatType2'),
+      substatValue2: relicForm.getFieldValue('substatValue2'),
+      substatType3: relicForm.getFieldValue('substatType3'),
+      substatValue3: relicForm.getFieldValue('substatValue3'),
+    }
+    if (compareRelics(props.selectedRelic, currentValue)) {
+      if (props.selectedRelic.verified) setHasChanged(true)
+      props.selectedRelic.verified = false
+      console.log('Relic unverified')
+    } else (setHasChanged(false))
```

Comment:
I think this is probably overcomplicating things, we dont need a state to track the status if we use the finalized relics to set the verified state in `onFinish`